template override colon bugfixes

corubba opened the pull request:
##### SUMMARY
ansible allows to specify environment overrides on the first line of a jinja2 template, which is a comma-separated list of colon-separated key-value-pairs. This PR fixes two issues with this feature.

When the user uses a wrong separator for a key-value-pair, ansible would previously fail with a undescriptive `ValueError` (see additional info below). This case is now explicitly checked and a meaningful message returned to the user. It however continues to be a "hard" error (so not catched by `ignore_errors`).

ansible would also split the key-value-pair on *all* colons, which results in a error (almost the same as the one above) and also prevents the usage of values containing colons. This no longer throws an error, ansible splits the override only on the first colon and passes all the rest on as-is as value.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
template

##### ADDITIONAL INFORMATION

template
```
#jinja2: lstrip_blocks=True
```
Error until now:
```
TASK [template]
*****************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: not enough values to unpack (expected 2, got 1)
fatal: [localhost]: FAILED! => {"changed": false, "msg": "ValueError: not enough values to unpack (expected 2, got 1)"}
```
Error now:
```
TASK [template]
*****************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleError: failed to parse jinja2 override 'lstrip_blocks=True'. Did you use something different from colon as key-value seperator?
fatal: [localhost]: FAILED! => {"changed": false, "msg": "AnsibleError: failed to parse jinja2 override 'lstrip_blocks=True'. Did you use something different from colon as key-value separator?"}
```

Akasurde commented on changelogs/fragments/template_override.yml:
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - More gracefully handle separator errors in jinja2 template overrides
```
```suggestion
  - More gracefully handle separator errors in jinja2 template overrides.
```

Akasurde commented on changelogs/fragments/template_override.yml:
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - More gracefully handle separator errors in jinja2 template overrides
+  - Support colons in jinja2 template override values
```
```suggestion
  - Support colons in jinja2 template override values.
```

corubba committed:
Proper error on missing jinja2 override separator

Properly catch the case when no (or a wrong) separator is used in a
jinja2 override, and return a useful error message to the user.

corubba committed:
Support colons in jinja2 override value

By limiting the split to 1, any colons in the value are preserved and
passed on.

corubba commented:
I applied the suggested changes to the changelog fragment, and added a link to the PR. Also rebased onto `devel` again since I was at it.
